android/src/main/kotlin/carnegietechnologies/gallery_saver/FileUtils.kt

```diff
@@ -261,4 +274,33 @@ internal object FileUtils {
         }
         return true
     }
+
+    private fun getAlbumFolderPath(folderName: String?, mediaType: MediaType): String {
+        var albumFolderPath: String = Environment.getExternalStorageDirectory().path
+        albumFolderPath = if (TextUtils.isEmpty(folderName)) {
+            val baseFolderName = if (mediaType == MediaType.image)
+                Environment.DIRECTORY_PICTURES else
+                Environment.DIRECTORY_MOVIES
+            createDirIfNotExist(
+                Environment.getExternalStoragePublicDirectory(baseFolderName).path
+            ) ?: albumFolderPath
+        } else {
+            createDirIfNotExist(albumFolderPath + File.separator + folderName)
+                ?: albumFolderPath
+        }
+        return albumFolderPath
+    }
+
+    private fun createDirIfNotExist(dirPath: String): String? {
+        val dir = File(dirPath)
+        if (!dir.exists()) {
+            if (dir.mkdirs()) {
+                return dir.path
+            } else {
+                return null
```

Comment:
should we throw an exception if mkdirs() returns false instead of returning null?

Reply:
No,we shouldn't. In that case Media file will be saved at "root" external dir.